src/transformers/testing_utils.py

```diff
@@ -1672,3 +1673,43 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def run_test_in_subprocess(test_case, target_func, inputs=None, timeout=600):
```

Comment:
Looks like this could be a decorator no? From your use later on you end up with public functions that just call the private ones.